### ISSUE TITLE
country and provider unknown

### ISSUE THREAD

**Issue by madsem** [matomo-org/tracker-proxy#28]
Since I switched to this proxy, providers and countries are all tracked as unknown.
Without using the tracker prxy it's working.

What is the issue here?

**Comment by madsem** [comment]
![screen shot 2018-02-17 at 4 29 41 pm](https://user-images.githubusercontent.com/3166991/36342522-43432402-1400-11e8-81b4-89762e3b78cf.png)

**Comment by sgiehl** [comment]
If you are using the Server module GeoIP location provider, the location data won't be available when using the tracker proxy. You may need to switch to PECL or PHP based detections.

**Comment by madsem** [comment]
Thank you for this info, this is indeed the case.
Sounds more like a bug to me though tbh?

I'm using it as NGINX module, GeoIP2.

It is def. available, Matomo just doesn't use it. I even mapped the $_SERVER vars to the ones Matomo uses internally.

**Comment by madsem** [comment]
okay bug was not correct, meant it more like a design flaw maybe.
It's still possible to override the remote_addr and get results that way for example.

But your answer makes sense, and good to know you're working on GeoIP2 also. Thanks!

**Comment by sgiehl** [comment]
Spoofing the senders IP address might not be something that should be done for HTTP requests.

Btw. It might not be good to use GeoIP2 server modules with Piwik already. GeoIP Legacy and GeoIP2 uses different codes for regions  (FIPS vs ISO). Currently Piwik is only able to handle the old ISO codes. 
Using GeoIP2 already might result in incorrect region data after GeoIP2 support is introduced in Piwik.

**Comment by madsem** [comment]
You're right Stefan. Just maybe also depends on the use case, just my opinion so that functionality doesn't break like in this case. Problem here is, our servers are set up this way, changing that would require a lot of work.

Thanks for the tip regarding regions, have to think how to solve this now. Need more coffee :)
